diff --git a/include/expr_ops.h b/include/expr_ops.h
index cecad95..0bbc0e1 100644
--- a/include/expr_ops.h
+++ b/include/expr_ops.h
@@ -17,7 +17,7 @@ struct expr_ops {
 	const void *(*get)(const struct nftnl_expr *e, uint16_t type, uint32_t *data_len);
 	int 	(*parse)(struct nftnl_expr *e, struct nlattr *attr);
 	void	(*build)(struct nlmsghdr *nlh, struct nftnl_expr *e);
-	int	(*snprintf)(char *buf, size_t len, uint32_t type, uint32_t flags, struct nftnl_expr *e);
+	int	(*snprintf_)(char *buf, size_t len, uint32_t type, uint32_t flags, struct nftnl_expr *e);
 	int	(*xml_parse)(struct nftnl_expr *e, mxml_node_t *tree,
 			     struct nftnl_parse_err *err);
 	int	(*json_parse)(struct nftnl_expr *e, json_t *data,
diff --git a/src/expr.c b/src/expr.c
index 3249a5c..1139875 100644
--- a/src/expr.c
+++ b/src/expr.c
@@ -268,7 +268,7 @@ int nftnl_expr_snprintf(char *buf, size_t size, struct nftnl_expr *expr,
 	int ret;
 	unsigned int offset = 0, len = size;
 
-	ret = expr->ops->snprintf(buf+offset, len, type, flags, expr);
+	ret = expr->ops->snprintf_(buf+offset, len, type, flags, expr);
 	SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
 	return offset;
diff --git a/src/expr/bitwise.c b/src/expr/bitwise.c
index a18aad5..ff095ad 100644
--- a/src/expr/bitwise.c
+++ b/src/expr/bitwise.c
@@ -318,7 +318,7 @@ struct expr_ops expr_ops_bitwise = {
 	.get		= nftnl_expr_bitwise_get,
 	.parse		= nftnl_expr_bitwise_parse,
 	.build		= nftnl_expr_bitwise_build,
-	.snprintf	= nftnl_expr_bitwise_snprintf,
+	.snprintf_	= nftnl_expr_bitwise_snprintf,
 	.xml_parse	= nftnl_expr_bitwise_xml_parse,
 	.json_parse	= nftnl_expr_bitwise_json_parse,
 };
diff --git a/src/expr/byteorder.c b/src/expr/byteorder.c
index f16bbf7..ba1ad55 100644
--- a/src/expr/byteorder.c
+++ b/src/expr/byteorder.c
@@ -334,7 +334,7 @@ struct expr_ops expr_ops_byteorder = {
 	.get		= nftnl_expr_byteorder_get,
 	.parse		= nftnl_expr_byteorder_parse,
 	.build		= nftnl_expr_byteorder_build,
-	.snprintf	= nftnl_expr_byteorder_snprintf,
+	.snprintf_	= nftnl_expr_byteorder_snprintf,
 	.xml_parse	= nftnl_expr_byteorder_xml_parse,
 	.json_parse	= nftnl_expr_byteorder_json_parse,
 };
diff --git a/src/expr/cmp.c b/src/expr/cmp.c
index 4125689..5c9c8b3 100644
--- a/src/expr/cmp.c
+++ b/src/expr/cmp.c
@@ -301,7 +301,7 @@ struct expr_ops expr_ops_cmp = {
 	.get		= nftnl_expr_cmp_get,
 	.parse		= nftnl_expr_cmp_parse,
 	.build		= nftnl_expr_cmp_build,
-	.snprintf	= nftnl_expr_cmp_snprintf,
+	.snprintf_	= nftnl_expr_cmp_snprintf,
 	.xml_parse	= nftnl_expr_cmp_xml_parse,
 	.json_parse	= nftnl_expr_cmp_json_parse,
 };
diff --git a/src/expr/counter.c b/src/expr/counter.c
index 4b5fb3f..092035f 100644
--- a/src/expr/counter.c
+++ b/src/expr/counter.c
@@ -206,7 +206,7 @@ struct expr_ops expr_ops_counter = {
 	.get		= nftnl_expr_counter_get,
 	.parse		= nftnl_expr_counter_parse,
 	.build		= nftnl_expr_counter_build,
-	.snprintf	= nftnl_expr_counter_snprintf,
+	.snprintf_	= nftnl_expr_counter_snprintf,
 	.xml_parse	= nftnl_expr_counter_xml_parse,
 	.json_parse	= nftnl_expr_counter_json_parse,
 };
diff --git a/src/expr/ct.c b/src/expr/ct.c
index 584dffe..55e3d39 100644
--- a/src/expr/ct.c
+++ b/src/expr/ct.c
@@ -377,7 +377,7 @@ struct expr_ops expr_ops_ct = {
 	.get		= nftnl_expr_ct_get,
 	.parse		= nftnl_expr_ct_parse,
 	.build		= nftnl_expr_ct_build,
-	.snprintf	= nftnl_expr_ct_snprintf,
+	.snprintf_	= nftnl_expr_ct_snprintf,
 	.xml_parse	= nftnl_expr_ct_xml_parse,
 	.json_parse	= nftnl_expr_ct_json_parse,
 };
diff --git a/src/expr/dynset.c b/src/expr/dynset.c
index ec8f2d3..62c01a7 100644
--- a/src/expr/dynset.c
+++ b/src/expr/dynset.c
@@ -369,7 +369,7 @@ struct expr_ops expr_ops_dynset = {
 	.get		= nftnl_expr_dynset_get,
 	.parse		= nftnl_expr_dynset_parse,
 	.build		= nftnl_expr_dynset_build,
-	.snprintf	= nftnl_expr_dynset_snprintf,
+	.snprintf_	= nftnl_expr_dynset_snprintf,
 	.xml_parse	= nftnl_expr_dynset_xml_parse,
 	.json_parse	= nftnl_expr_dynset_json_parse,
 };
diff --git a/src/expr/exthdr.c b/src/expr/exthdr.c
index 2e9b0b0..d97d34d 100644
--- a/src/expr/exthdr.c
+++ b/src/expr/exthdr.c
@@ -312,7 +312,7 @@ struct expr_ops expr_ops_exthdr = {
 	.get		= nftnl_expr_exthdr_get,
 	.parse		= nftnl_expr_exthdr_parse,
 	.build		= nftnl_expr_exthdr_build,
-	.snprintf	= nftnl_expr_exthdr_snprintf,
+	.snprintf_	= nftnl_expr_exthdr_snprintf,
 	.xml_parse	= nftnl_expr_exthdr_xml_parse,
 	.json_parse	= nftnl_expr_exthdr_json_parse,
 };
diff --git a/src/expr/immediate.c b/src/expr/immediate.c
index c2fb9ad..921d9db 100644
--- a/src/expr/immediate.c
+++ b/src/expr/immediate.c
@@ -326,7 +326,7 @@ struct expr_ops expr_ops_immediate = {
 	.get		= nftnl_expr_immediate_get,
 	.parse		= nftnl_expr_immediate_parse,
 	.build		= nftnl_expr_immediate_build,
-	.snprintf	= nftnl_expr_immediate_snprintf,
+	.snprintf_	= nftnl_expr_immediate_snprintf,
 	.xml_parse	= nftnl_expr_immediate_xml_parse,
 	.json_parse	= nftnl_expr_immediate_json_parse,
 };
diff --git a/src/expr/limit.c b/src/expr/limit.c
index e84745d..81a0c61 100644
--- a/src/expr/limit.c
+++ b/src/expr/limit.c
@@ -216,7 +216,7 @@ struct expr_ops expr_ops_limit = {
 	.get		= nftnl_expr_limit_get,
 	.parse		= nftnl_expr_limit_parse,
 	.build		= nftnl_expr_limit_build,
-	.snprintf	= nftnl_expr_limit_snprintf,
+	.snprintf_	= nftnl_expr_limit_snprintf,
 	.xml_parse	= nftnl_expr_limit_xml_parse,
 	.json_parse	= nftnl_expr_limit_json_parse,
 };
diff --git a/src/expr/log.c b/src/expr/log.c
index f615d8e..a6f16b0 100644
--- a/src/expr/log.c
+++ b/src/expr/log.c
@@ -341,7 +341,7 @@ struct expr_ops expr_ops_log = {
 	.get		= nftnl_expr_log_get,
 	.parse		= nftnl_expr_log_parse,
 	.build		= nftnl_expr_log_build,
-	.snprintf	= nftnl_expr_log_snprintf,
+	.snprintf_	= nftnl_expr_log_snprintf,
 	.xml_parse	= nftnl_expr_log_xml_parse,
 	.json_parse	= nftnl_expr_log_json_parse,
 };
diff --git a/src/expr/lookup.c b/src/expr/lookup.c
index d911cb6..5c01e94 100644
--- a/src/expr/lookup.c
+++ b/src/expr/lookup.c
@@ -266,7 +266,7 @@ struct expr_ops expr_ops_lookup = {
 	.get		= nftnl_expr_lookup_get,
 	.parse		= nftnl_expr_lookup_parse,
 	.build		= nftnl_expr_lookup_build,
-	.snprintf	= nftnl_expr_lookup_snprintf,
+	.snprintf_	= nftnl_expr_lookup_snprintf,
 	.xml_parse	= nftnl_expr_lookup_xml_parse,
 	.json_parse	= nftnl_expr_lookup_json_parse,
 };
diff --git a/src/expr/masq.c b/src/expr/masq.c
index 01512b4..d5cb9cb 100644
--- a/src/expr/masq.c
+++ b/src/expr/masq.c
@@ -180,7 +180,7 @@ struct expr_ops expr_ops_masq = {
 	.get		= nftnl_expr_masq_get,
 	.parse		= nftnl_expr_masq_parse,
 	.build		= nftnl_expr_masq_build,
-	.snprintf	= nftnl_expr_masq_snprintf,
+	.snprintf_	= nftnl_expr_masq_snprintf,
 	.xml_parse	= nftnl_expr_masq_xml_parse,
 	.json_parse	= nftnl_expr_masq_json_parse,
 };
diff --git a/src/expr/match.c b/src/expr/match.c
index ddecb43..8a29f1a 100644
--- a/src/expr/match.c
+++ b/src/expr/match.c
@@ -249,7 +249,7 @@ struct expr_ops expr_ops_match = {
 	.get		= nftnl_expr_match_get,
 	.parse		= nftnl_expr_match_parse,
 	.build		= nftnl_expr_match_build,
-	.snprintf	= nftnl_expr_match_snprintf,
+	.snprintf_	= nftnl_expr_match_snprintf,
 	.xml_parse 	= nftnl_expr_match_xml_parse,
 	.json_parse 	= nftnl_expr_match_json_parse,
 };
diff --git a/src/expr/meta.c b/src/expr/meta.c
index 546b225..a9329c8 100644
--- a/src/expr/meta.c
+++ b/src/expr/meta.c
@@ -303,7 +303,7 @@ struct expr_ops expr_ops_meta = {
 	.get		= nftnl_expr_meta_get,
 	.parse		= nftnl_expr_meta_parse,
 	.build		= nftnl_expr_meta_build,
-	.snprintf	= nftnl_expr_meta_snprintf,
+	.snprintf_	= nftnl_expr_meta_snprintf,
 	.xml_parse 	= nftnl_expr_meta_xml_parse,
 	.json_parse 	= nftnl_expr_meta_json_parse,
 };
diff --git a/src/expr/nat.c b/src/expr/nat.c
index 421983f..fe21561 100644
--- a/src/expr/nat.c
+++ b/src/expr/nat.c
@@ -412,7 +412,7 @@ struct expr_ops expr_ops_nat = {
 	.get		= nftnl_expr_nat_get,
 	.parse		= nftnl_expr_nat_parse,
 	.build		= nftnl_expr_nat_build,
-	.snprintf	= nftnl_expr_nat_snprintf,
+	.snprintf_	= nftnl_expr_nat_snprintf,
 	.xml_parse	= nftnl_expr_nat_xml_parse,
 	.json_parse	= nftnl_expr_nat_json_parse,
 };
diff --git a/src/expr/payload.c b/src/expr/payload.c
index be10663..4031782 100644
--- a/src/expr/payload.c
+++ b/src/expr/payload.c
@@ -293,7 +293,7 @@ struct expr_ops expr_ops_payload = {
 	.get		= nftnl_expr_payload_get,
 	.parse		= nftnl_expr_payload_parse,
 	.build		= nftnl_expr_payload_build,
-	.snprintf	= nftnl_expr_payload_snprintf,
+	.snprintf_	= nftnl_expr_payload_snprintf,
 	.xml_parse	= nftnl_expr_payload_xml_parse,
 	.json_parse	= nftnl_expr_payload_json_parse,
 };
diff --git a/src/expr/queue.c b/src/expr/queue.c
index 21126ed..b430e39 100644
--- a/src/expr/queue.c
+++ b/src/expr/queue.c
@@ -251,7 +251,7 @@ struct expr_ops expr_ops_queue = {
 	.get		= nftnl_expr_queue_get,
 	.parse		= nftnl_expr_queue_parse,
 	.build		= nftnl_expr_queue_build,
-	.snprintf	= nftnl_expr_queue_snprintf,
+	.snprintf_	= nftnl_expr_queue_snprintf,
 	.xml_parse	= nftnl_expr_queue_xml_parse,
 	.json_parse	= nftnl_expr_queue_json_parse,
 };
diff --git a/src/expr/redir.c b/src/expr/redir.c
index b8fb1fe..fa309fc 100644
--- a/src/expr/redir.c
+++ b/src/expr/redir.c
@@ -250,7 +250,7 @@ struct expr_ops expr_ops_redir = {
 	.get		= nftnl_expr_redir_get,
 	.parse		= nftnl_expr_redir_parse,
 	.build		= nftnl_expr_redir_build,
-	.snprintf	= nftnl_expr_redir_snprintf,
+	.snprintf_	= nftnl_expr_redir_snprintf,
 	.xml_parse	= nftnl_expr_redir_xml_parse,
 	.json_parse	= nftnl_expr_redir_json_parse,
 };
diff --git a/src/expr/reject.c b/src/expr/reject.c
index 979817a..6b1f773 100644
--- a/src/expr/reject.c
+++ b/src/expr/reject.c
@@ -207,7 +207,7 @@ struct expr_ops expr_ops_reject = {
 	.get		= nftnl_expr_reject_get,
 	.parse		= nftnl_expr_reject_parse,
 	.build		= nftnl_expr_reject_build,
-	.snprintf	= nftnl_expr_reject_snprintf,
+	.snprintf_	= nftnl_expr_reject_snprintf,
 	.xml_parse	= nftnl_expr_reject_xml_parse,
 	.json_parse	= nftnl_expr_reject_json_parse,
 };
diff --git a/src/expr/target.c b/src/expr/target.c
index cbbee60..62ecf4a 100644
--- a/src/expr/target.c
+++ b/src/expr/target.c
@@ -250,7 +250,7 @@ struct expr_ops expr_ops_target = {
 	.get		= nftnl_expr_target_get,
 	.parse		= nftnl_expr_target_parse,
 	.build		= nftnl_expr_target_build,
-	.snprintf	= nftnl_expr_target_snprintf,
+	.snprintf_	= nftnl_expr_target_snprintf,
 	.xml_parse	= nftnl_expr_target_xml_parse,
 	.json_parse	= nftnl_expr_target_json_parse,
 };
diff --git a/src/rule.c b/src/rule.c
index 0fb27ed..cfc75f5 100644
--- a/src/rule.c
+++ b/src/rule.c
@@ -765,7 +765,7 @@ static int nftnl_rule_snprintf_json(char *buf, size_t size, struct nftnl_rule *r
 			       "{\"type\":\"%s\",", expr->ops->name);
 		SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
-		ret = expr->ops->snprintf(buf+offset, len, type, flags, expr);
+		ret = expr->ops->snprintf_(buf+offset, len, type, flags, expr);
 		SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
 		/*
