diff --git a/include/expr_ops.h b/include/expr_ops.h
index 3c0cb18..f3ebef0 100644
--- a/include/expr_ops.h
+++ b/include/expr_ops.h
@@ -17,7 +17,7 @@ struct expr_ops {
 	const void *(*get)(const struct nftnl_expr *e, uint16_t type, uint32_t *data_len);
 	int 	(*parse)(struct nftnl_expr *e, struct nlattr *attr);
 	void	(*build)(struct nlmsghdr *nlh, const struct nftnl_expr *e);
-	int	(*snprintf)(char *buf, size_t len, uint32_t type, uint32_t flags, const struct nftnl_expr *e);
+	int	(*snprintf_)(char *buf, size_t len, uint32_t type, uint32_t flags, const struct nftnl_expr *e);
 	int	(*xml_parse)(struct nftnl_expr *e, mxml_node_t *tree,
 			     struct nftnl_parse_err *err);
 	int	(*json_parse)(struct nftnl_expr *e, json_t *data,
diff --git a/src/expr.c b/src/expr.c
index ed07dc4..b07337c 100644
--- a/src/expr.c
+++ b/src/expr.c
@@ -268,7 +268,7 @@ int nftnl_expr_snprintf(char *buf, size_t size, const struct nftnl_expr *expr,
 	int ret;
 	unsigned int offset = 0, len = size;
 
-	ret = expr->ops->snprintf(buf+offset, len, type, flags, expr);
+	ret = expr->ops->snprintf_(buf+offset, len, type, flags, expr);
 	SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
 	return offset;
diff --git a/src/expr/bitwise.c b/src/expr/bitwise.c
index 2fd4b74..532726d 100644
--- a/src/expr/bitwise.c
+++ b/src/expr/bitwise.c
@@ -318,7 +318,7 @@ struct expr_ops expr_ops_bitwise = {
 	.get		= nftnl_expr_bitwise_get,
 	.parse		= nftnl_expr_bitwise_parse,
 	.build		= nftnl_expr_bitwise_build,
-	.snprintf	= nftnl_expr_bitwise_snprintf,
+	.snprintf_	= nftnl_expr_bitwise_snprintf,
 	.xml_parse	= nftnl_expr_bitwise_xml_parse,
 	.json_parse	= nftnl_expr_bitwise_json_parse,
 };
diff --git a/src/expr/byteorder.c b/src/expr/byteorder.c
index ca697cf..3423a3e 100644
--- a/src/expr/byteorder.c
+++ b/src/expr/byteorder.c
@@ -334,7 +334,7 @@ struct expr_ops expr_ops_byteorder = {
 	.get		= nftnl_expr_byteorder_get,
 	.parse		= nftnl_expr_byteorder_parse,
 	.build		= nftnl_expr_byteorder_build,
-	.snprintf	= nftnl_expr_byteorder_snprintf,
+	.snprintf_	= nftnl_expr_byteorder_snprintf,
 	.xml_parse	= nftnl_expr_byteorder_xml_parse,
 	.json_parse	= nftnl_expr_byteorder_json_parse,
 };
diff --git a/src/expr/cmp.c b/src/expr/cmp.c
index f3dd62c..27230a1 100644
--- a/src/expr/cmp.c
+++ b/src/expr/cmp.c
@@ -301,7 +301,7 @@ struct expr_ops expr_ops_cmp = {
 	.get		= nftnl_expr_cmp_get,
 	.parse		= nftnl_expr_cmp_parse,
 	.build		= nftnl_expr_cmp_build,
-	.snprintf	= nftnl_expr_cmp_snprintf,
+	.snprintf_	= nftnl_expr_cmp_snprintf,
 	.xml_parse	= nftnl_expr_cmp_xml_parse,
 	.json_parse	= nftnl_expr_cmp_json_parse,
 };
diff --git a/src/expr/counter.c b/src/expr/counter.c
index 1f818c4..b1f123e 100644
--- a/src/expr/counter.c
+++ b/src/expr/counter.c
@@ -206,7 +206,7 @@ struct expr_ops expr_ops_counter = {
 	.get		= nftnl_expr_counter_get,
 	.parse		= nftnl_expr_counter_parse,
 	.build		= nftnl_expr_counter_build,
-	.snprintf	= nftnl_expr_counter_snprintf,
+	.snprintf_	= nftnl_expr_counter_snprintf,
 	.xml_parse	= nftnl_expr_counter_xml_parse,
 	.json_parse	= nftnl_expr_counter_json_parse,
 };
diff --git a/src/expr/ct.c b/src/expr/ct.c
index 7d96df4..2b3fd91 100644
--- a/src/expr/ct.c
+++ b/src/expr/ct.c
@@ -381,7 +381,7 @@ struct expr_ops expr_ops_ct = {
 	.get		= nftnl_expr_ct_get,
 	.parse		= nftnl_expr_ct_parse,
 	.build		= nftnl_expr_ct_build,
-	.snprintf	= nftnl_expr_ct_snprintf,
+	.snprintf_	= nftnl_expr_ct_snprintf,
 	.xml_parse	= nftnl_expr_ct_xml_parse,
 	.json_parse	= nftnl_expr_ct_json_parse,
 };
diff --git a/src/expr/dup.c b/src/expr/dup.c
index 694db32..c7435c5 100644
--- a/src/expr/dup.c
+++ b/src/expr/dup.c
@@ -210,7 +210,7 @@ struct expr_ops expr_ops_dup = {
 	.get		= nftnl_expr_dup_get,
 	.parse		= nftnl_expr_dup_parse,
 	.build		= nftnl_expr_dup_build,
-	.snprintf	= nftnl_expr_dup_snprintf,
+	.snprintf_	= nftnl_expr_dup_snprintf,
 	.xml_parse	= nftnl_expr_dup_xml_parse,
 	.json_parse	= nftnl_expr_dup_json_parse,
 };
diff --git a/src/expr/dynset.c b/src/expr/dynset.c
index c8d97a5..dca0111 100644
--- a/src/expr/dynset.c
+++ b/src/expr/dynset.c
@@ -373,7 +373,7 @@ struct expr_ops expr_ops_dynset = {
 	.get		= nftnl_expr_dynset_get,
 	.parse		= nftnl_expr_dynset_parse,
 	.build		= nftnl_expr_dynset_build,
-	.snprintf	= nftnl_expr_dynset_snprintf,
+	.snprintf_	= nftnl_expr_dynset_snprintf,
 	.xml_parse	= nftnl_expr_dynset_xml_parse,
 	.json_parse	= nftnl_expr_dynset_json_parse,
 };
diff --git a/src/expr/exthdr.c b/src/expr/exthdr.c
index 6489a48..e11573a 100644
--- a/src/expr/exthdr.c
+++ b/src/expr/exthdr.c
@@ -312,7 +312,7 @@ struct expr_ops expr_ops_exthdr = {
 	.get		= nftnl_expr_exthdr_get,
 	.parse		= nftnl_expr_exthdr_parse,
 	.build		= nftnl_expr_exthdr_build,
-	.snprintf	= nftnl_expr_exthdr_snprintf,
+	.snprintf_	= nftnl_expr_exthdr_snprintf,
 	.xml_parse	= nftnl_expr_exthdr_xml_parse,
 	.json_parse	= nftnl_expr_exthdr_json_parse,
 };
diff --git a/src/expr/fwd.c b/src/expr/fwd.c
index 8fa6d66..60f7e1e 100644
--- a/src/expr/fwd.c
+++ b/src/expr/fwd.c
@@ -183,7 +183,7 @@ struct expr_ops expr_ops_fwd = {
 	.get		= nftnl_expr_fwd_get,
 	.parse		= nftnl_expr_fwd_parse,
 	.build		= nftnl_expr_fwd_build,
-	.snprintf	= nftnl_expr_fwd_snprintf,
+	.snprintf_	= nftnl_expr_fwd_snprintf,
 	.xml_parse	= nftnl_expr_fwd_xml_parse,
 	.json_parse	= nftnl_expr_fwd_json_parse,
 };
diff --git a/src/expr/immediate.c b/src/expr/immediate.c
index eb2ca0f..9275095 100644
--- a/src/expr/immediate.c
+++ b/src/expr/immediate.c
@@ -327,7 +327,7 @@ struct expr_ops expr_ops_immediate = {
 	.get		= nftnl_expr_immediate_get,
 	.parse		= nftnl_expr_immediate_parse,
 	.build		= nftnl_expr_immediate_build,
-	.snprintf	= nftnl_expr_immediate_snprintf,
+	.snprintf_	= nftnl_expr_immediate_snprintf,
 	.xml_parse	= nftnl_expr_immediate_xml_parse,
 	.json_parse	= nftnl_expr_immediate_json_parse,
 };
diff --git a/src/expr/limit.c b/src/expr/limit.c
index 4bd096e..5c728c2 100644
--- a/src/expr/limit.c
+++ b/src/expr/limit.c
@@ -296,7 +296,7 @@ struct expr_ops expr_ops_limit = {
 	.get		= nftnl_expr_limit_get,
 	.parse		= nftnl_expr_limit_parse,
 	.build		= nftnl_expr_limit_build,
-	.snprintf	= nftnl_expr_limit_snprintf,
+	.snprintf_	= nftnl_expr_limit_snprintf,
 	.xml_parse	= nftnl_expr_limit_xml_parse,
 	.json_parse	= nftnl_expr_limit_json_parse,
 };
diff --git a/src/expr/log.c b/src/expr/log.c
index c3dc0a6..620fb54 100644
--- a/src/expr/log.c
+++ b/src/expr/log.c
@@ -341,7 +341,7 @@ struct expr_ops expr_ops_log = {
 	.get		= nftnl_expr_log_get,
 	.parse		= nftnl_expr_log_parse,
 	.build		= nftnl_expr_log_build,
-	.snprintf	= nftnl_expr_log_snprintf,
+	.snprintf_	= nftnl_expr_log_snprintf,
 	.xml_parse	= nftnl_expr_log_xml_parse,
 	.json_parse	= nftnl_expr_log_json_parse,
 };
diff --git a/src/expr/lookup.c b/src/expr/lookup.c
index ed32ba6..0fe23af 100644
--- a/src/expr/lookup.c
+++ b/src/expr/lookup.c
@@ -270,7 +270,7 @@ struct expr_ops expr_ops_lookup = {
 	.get		= nftnl_expr_lookup_get,
 	.parse		= nftnl_expr_lookup_parse,
 	.build		= nftnl_expr_lookup_build,
-	.snprintf	= nftnl_expr_lookup_snprintf,
+	.snprintf_	= nftnl_expr_lookup_snprintf,
 	.xml_parse	= nftnl_expr_lookup_xml_parse,
 	.json_parse	= nftnl_expr_lookup_json_parse,
 };
diff --git a/src/expr/masq.c b/src/expr/masq.c
index 7296590..c6fe567 100644
--- a/src/expr/masq.c
+++ b/src/expr/masq.c
@@ -236,7 +236,7 @@ struct expr_ops expr_ops_masq = {
 	.get		= nftnl_expr_masq_get,
 	.parse		= nftnl_expr_masq_parse,
 	.build		= nftnl_expr_masq_build,
-	.snprintf	= nftnl_expr_masq_snprintf,
+	.snprintf_	= nftnl_expr_masq_snprintf,
 	.xml_parse	= nftnl_expr_masq_xml_parse,
 	.json_parse	= nftnl_expr_masq_json_parse,
 };
diff --git a/src/expr/match.c b/src/expr/match.c
index 2929b43..fda001f 100644
--- a/src/expr/match.c
+++ b/src/expr/match.c
@@ -249,7 +249,7 @@ struct expr_ops expr_ops_match = {
 	.get		= nftnl_expr_match_get,
 	.parse		= nftnl_expr_match_parse,
 	.build		= nftnl_expr_match_build,
-	.snprintf	= nftnl_expr_match_snprintf,
+	.snprintf_	= nftnl_expr_match_snprintf,
 	.xml_parse 	= nftnl_expr_match_xml_parse,
 	.json_parse 	= nftnl_expr_match_json_parse,
 };
diff --git a/src/expr/meta.c b/src/expr/meta.c
index a478a89..3ca97e9 100644
--- a/src/expr/meta.c
+++ b/src/expr/meta.c
@@ -304,7 +304,7 @@ struct expr_ops expr_ops_meta = {
 	.get		= nftnl_expr_meta_get,
 	.parse		= nftnl_expr_meta_parse,
 	.build		= nftnl_expr_meta_build,
-	.snprintf	= nftnl_expr_meta_snprintf,
+	.snprintf_	= nftnl_expr_meta_snprintf,
 	.xml_parse 	= nftnl_expr_meta_xml_parse,
 	.json_parse 	= nftnl_expr_meta_json_parse,
 };
diff --git a/src/expr/nat.c b/src/expr/nat.c
index 1ae7f77..f4bac92 100644
--- a/src/expr/nat.c
+++ b/src/expr/nat.c
@@ -412,7 +412,7 @@ struct expr_ops expr_ops_nat = {
 	.get		= nftnl_expr_nat_get,
 	.parse		= nftnl_expr_nat_parse,
 	.build		= nftnl_expr_nat_build,
-	.snprintf	= nftnl_expr_nat_snprintf,
+	.snprintf_	= nftnl_expr_nat_snprintf,
 	.xml_parse	= nftnl_expr_nat_xml_parse,
 	.json_parse	= nftnl_expr_nat_json_parse,
 };
diff --git a/src/expr/payload.c b/src/expr/payload.c
index b4fd9c5..b4720fd 100644
--- a/src/expr/payload.c
+++ b/src/expr/payload.c
@@ -343,7 +343,7 @@ struct expr_ops expr_ops_payload = {
 	.get		= nftnl_expr_payload_get,
 	.parse		= nftnl_expr_payload_parse,
 	.build		= nftnl_expr_payload_build,
-	.snprintf	= nftnl_expr_payload_snprintf,
+	.snprintf_	= nftnl_expr_payload_snprintf,
 	.xml_parse	= nftnl_expr_payload_xml_parse,
 	.json_parse	= nftnl_expr_payload_json_parse,
 };
diff --git a/src/expr/queue.c b/src/expr/queue.c
index a2e6ed5..f847fa4 100644
--- a/src/expr/queue.c
+++ b/src/expr/queue.c
@@ -251,7 +251,7 @@ struct expr_ops expr_ops_queue = {
 	.get		= nftnl_expr_queue_get,
 	.parse		= nftnl_expr_queue_parse,
 	.build		= nftnl_expr_queue_build,
-	.snprintf	= nftnl_expr_queue_snprintf,
+	.snprintf_	= nftnl_expr_queue_snprintf,
 	.xml_parse	= nftnl_expr_queue_xml_parse,
 	.json_parse	= nftnl_expr_queue_json_parse,
 };
diff --git a/src/expr/redir.c b/src/expr/redir.c
index c6eedfb..55764bd 100644
--- a/src/expr/redir.c
+++ b/src/expr/redir.c
@@ -250,7 +250,7 @@ struct expr_ops expr_ops_redir = {
 	.get		= nftnl_expr_redir_get,
 	.parse		= nftnl_expr_redir_parse,
 	.build		= nftnl_expr_redir_build,
-	.snprintf	= nftnl_expr_redir_snprintf,
+	.snprintf_	= nftnl_expr_redir_snprintf,
 	.xml_parse	= nftnl_expr_redir_xml_parse,
 	.json_parse	= nftnl_expr_redir_json_parse,
 };
diff --git a/src/expr/reject.c b/src/expr/reject.c
index 1953328..dcf9883 100644
--- a/src/expr/reject.c
+++ b/src/expr/reject.c
@@ -207,7 +207,7 @@ struct expr_ops expr_ops_reject = {
 	.get		= nftnl_expr_reject_get,
 	.parse		= nftnl_expr_reject_parse,
 	.build		= nftnl_expr_reject_build,
-	.snprintf	= nftnl_expr_reject_snprintf,
+	.snprintf_	= nftnl_expr_reject_snprintf,
 	.xml_parse	= nftnl_expr_reject_xml_parse,
 	.json_parse	= nftnl_expr_reject_json_parse,
 };
diff --git a/src/expr/target.c b/src/expr/target.c
index 68a7d8a..f618d84 100644
--- a/src/expr/target.c
+++ b/src/expr/target.c
@@ -250,7 +250,7 @@ struct expr_ops expr_ops_target = {
 	.get		= nftnl_expr_target_get,
 	.parse		= nftnl_expr_target_parse,
 	.build		= nftnl_expr_target_build,
-	.snprintf	= nftnl_expr_target_snprintf,
+	.snprintf_	= nftnl_expr_target_snprintf,
 	.xml_parse	= nftnl_expr_target_xml_parse,
 	.json_parse	= nftnl_expr_target_json_parse,
 };
diff --git a/src/rule.c b/src/rule.c
index 8ee8648..d55ef2c 100644
--- a/src/rule.c
+++ b/src/rule.c
@@ -775,7 +775,7 @@ static int nftnl_rule_snprintf_json(char *buf, size_t size,
 			       "{\"type\":\"%s\",", expr->ops->name);
 		SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
-		ret = expr->ops->snprintf(buf+offset, len, type, flags, expr);
+		ret = expr->ops->snprintf_(buf+offset, len, type, flags, expr);
 		SNPRINTF_BUFFER_SIZE(ret, size, len, offset);
 
 		/*
